src/sentry/runner/commands/devservices.py

```diff
@@ -13,15 +18,34 @@
     os.environ["DOCKER_HOST"] = "unix://" + RAW_SOCKET_HACK_PATH
 
 
-def get_docker_client():
-    import docker
-
+def get_docker_client() -> docker.DockerClient:
     client = docker.from_env()
     try:
         client.ping()
         return client
-    except Exception:
-        raise click.ClickException("Make sure Docker is running.")
+    except requests.exceptions.ConnectionError:
+        click.echo("Attempting to start docker...")
+        if sys.platform == "darwin":
+            subprocess.check_call(
+                ("open", "-a", "/Applications/Docker.app", "--args", "--unattended")
+            )
+        else:
+            click.echo("Unable to start docker.")
+            raise click.ClickException("Make sure docker is running.")
+
+        max_wait = 60
+        start = time.monotonic()
+
+        click.echo(f"Waiting for docker to be ready.... (timeout in {max_wait}s)")
+        while time.monotonic() - start < max_wait:
+            time.sleep(1)
+            try:
+                client.ping()
+                return client
+            except (requests.exceptions.ConnectionError, docker.errors.APIError):
```

Comment:
should these errors be the same as above?

Reply:
I can make them the same, but what I found what that the docker error only appeared while docker was starting. 

My hunch is connection error means there is nothing to connect to (i.e. docker is not running) and in the latter case docker has started but it's unable to handle the connection request.

Reply:
ah interesting -- I'd probably make them the same (or document it, but probably simpler to make them the same)